src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java

```diff
@@ -347,6 +352,16 @@ public void checkIfSelfOptOrAdmin(String currentUser, String identifier) {
         }
     }
 
+    public boolean hasOptedIn(String groupingPath, String uhIdentifier) {
+        GroupingOptAttributes groupingOptAttributes = groupingOwnerService.groupingOptAttributes(uhIdentifier, groupingPath);
+        return groupingOptAttributes.isOptInOn();
+    }
+
+    public boolean hasOptedOut(String groupingPath, String uhIdentifier) {
+        GroupingOptAttributes groupingOptAttributes = groupingOwnerService.groupingOptAttributes(uhIdentifier, groupingPath);
+        return groupingOptAttributes.isOptOutOn();
+    }
```

Comment:
These two functions would be better suited inside GroupingOwnerService because they check if a grouping has optIn or optOut enabled, which only a grouping owner should know the status of. Then function names should be isOptInOn and isOptOutOn since the verb "has" implies that a user has recently opted in, which is not what the function does. The function checks if the grouping you own has optIn or optOut enabled. Also an integration test is needed.

Reply:
Jordan, sorry for confusion.